test/inductor/test_compiled_autograd.py

```diff
@@ -2091,17 +2097,23 @@ def fn():
 
         self.check_output_and_recompiles(fn, 3)
 
-    @unittest.skipIf(not HAS_CUDA, "requires cuda")
+    @unittest.skipIf(not HAS_GPU, "requires gpu")
     def test_free_activation_memory(self):
         script = """
 import torch
+from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_CUDA, HAS_XPU
 
 def main():
-    assert(torch.cuda.memory_allocated() == 0)
+    device_module = None
+    if HAS_CUDA:
```

Comment:
please extend the general device_interface in https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/device_interface.py
and use `device_interface.memory_allocated`